ci/create_database.py

```diff
@@ -89,79 +85,54 @@ async def create_database():
             assert len(rows) == 1
             return
 
-    with open(create_database_config['admin_password_file'], encoding='utf-8') as f:
-        admin_password = f.read()
-
-    with open(create_database_config['user_password_file'], encoding='utf-8') as f:
-        user_password = f.read()
-
-    admin_exists = await db.execute_and_fetchone("SELECT user FROM mysql.user WHERE user=%s", (admin_username,))
-    admin_exists = admin_exists and admin_exists.get('user') == admin_username
-
-    user_exists = await db.execute_and_fetchone("SELECT user FROM mysql.user WHERE user=%s", (user_username,))
-    user_exists = user_exists and user_exists.get('user') == user_username
-
-    create_admin_or_alter_password = (
-        f"CREATE USER '{admin_username}'@'%' IDENTIFIED BY '{admin_password}';"
-        if not admin_exists
-        else "ALTER USER '{admin_username}'@'%' IDENTIFIED BY '{admin_password}';"
-    )
+    async def create_user_idempotent(admin_or_user, mysql_username, mysql_password):
```

Comment:
This isn't really idempotent because someone could race between your SELECT and the CREATE USER. I'm not sure that matters though. There's only one person creating users at any time, right?

Reply:
Well, I guess what I mean to say is: it *is* single-writer idempotent but not *atomic* so with multiple writers its not idempotent.

Reply:
Good point, I think I'm abusing idempotent a little bit here. What I really mean is just `create_user_if_doesnt_exist`.